src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -141,7 +142,7 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     tests.Add(test);
             }
 
-            return tests.Count > 0
+            return tests.Count > 0 || builders.OfType<TestCaseSourceAttribute>().Any()
```

Comment:
Because all these tests have `CombinatorialAttribute`.
Oh! Lol! We're getting nailed by https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs#L136.

`builders.Count` will just have to be checked before `CombinatorialAttribute` is added to the builders collection.